wait-for.c

```diff
@@ -270,15 +268,18 @@ int main(int argc, const char **argv) {
 		goto exit;
 	}
 
+	dirpath_buffer = strdup(extrav[0]);
+	if (!dirpath_buffer) {
+		perror("could not create working buffer for given path");
+		status = 1;
+		goto exit;
+	}
+
+	const char *dirpath = dirname(dirpath_buffer); // Always returns a valid pointer
```

Comment:
Instead of the comment, please `#include <assert.c>` and `assert()` that that's the case.